Connection/Connection.php

```diff
@@ -650,7 +656,14 @@ public function isRemoteAddressConsidered()
      */
     public function getRemoteAddress()
     {
-        return $this->_remoteAddress;
+        $stream   = $this->getStream();
+        $streamId = (int) $stream;
```

Comment:
Are we sure this is unique? Maybe `spl_object_hash` would be a better alternative?